### PR TITLE
Add CLI parameters for disable/enable peer

### PR DESCRIPTION
**Pull request by linuskendall**
Adds a --name parameter and --yes parameter to the enable-peer/disable-peer. Helps with large network management where we need some automated tools to disable/enable peers.

Fixes tonarino/innernet#214.

### CODE REVIEW

File: shared/src/prompts.rs
Comment (strohel) [inline shared/src/prompts.rs:421]: Super-nit/curiosity question: previously `peer` was `&Peer`, now we've changed it to `Peer` (if I'm looking correctly), needing to `clone()`. Can we keep it a `&Peer`, perhaps if we change `.into_iter()` to `.iter()` in the first arm?

**Review by strohel — APPROVED** [review]
Looking good, thanks for your contribution! Added just one supernit/question.

I'd wait a couple of days to that others also have an opportunity to review, but then this is good for merging from me.

File: shared/src/types.rs
Comment (bschwind) [inline shared/src/types.rs:360]: Am I correct in that the `yes` flag can only really be effectively used alongside the `name` flag? In that case we may want to add a [`requires`](https://docs.rs/clap/4.1.6/clap/builder/struct.Arg.html#method.requires) dependency, though I'm not sure if we can express that with the clap-derive syntax.

**Review by bschwind — APPROVED** [review]
Overall this looks good, thanks for the PR! I think this is good to merge after following up on the clarifying questions.

**Comment by linuskendall** [comment]
> Overall this looks good, thanks for the PR! I think this is good to merge after following up on the clarifying questions.

Thanks for the quick review - I have been travelogn but will take a look and address the comments next week.

File: shared/src/types.rs
Comment (goodhoko) [reply shared/src/types.rs:360]: Per the [docs](https://docs.rs/clap/latest/clap/_derive/#terminology) it should be doable with a mere

```rust 
    #[arg(long, requires("name"))]
    pub yes: bool,
```

File: shared/src/prompts.rs
Comment (linuskendall) [reply shared/src/prompts.rs]: It looks like those clones were just superfluous. Removed.

File: shared/src/types.rs
Comment (linuskendall) [reply shared/src/types.rs:359]: Thanks, I made the change accordingly.

**Comment by linuskendall** [comment]
> Overall this looks good, thanks for the PR! I think this is good to merge after following up on the clarifying questions.

Sorry for leaving this dangling for so long! I just lost of track of it. Updated as per the comments.

**Review by strohel — APPROVED** [review]
Looking great, thanks, @linuskendall! Good to merge, though would be good to test in real life first.

Want anyone in the community to quickly test? Ideally both the server and client variants of the new options.

**Comment by linuskendall** [comment]
Good to test. For reference - we have been running this patch for the last
6 months or so on our innernet instance!

(Deployed it and kind of forgot that it was a patch we added, so forgot
about this PR! Sorry!)

**Comment by goodhoko** [comment]
@strohel Taking you attention back to this. I think this could me deemed as tested. Can we merge?

**Comment by strohel** [comment]
Thanks for ping, I've hit the merge button. Thanks everybody contributing ans sorry for the delays!

**Pull request merged**
